### PR TITLE
Optimize tc-opt

### PR DESCRIPTION
**Pull request by eeeqeee**
## Describe your changes

Only check feasiblity when `result > max`

## Issue or discord link

-

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code, in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] Ran `yarn run mini-ci` locally to validate format + lint.
- [ ] If there were format issues, I ran `nx format write` to resolve them automatically.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts [hunk 1]
```diff
@@ -237,6 +220,23 @@ export function optimizeTcUsingNodes(
       }
       const result = results[0]
```
Comment (frzyc) [inline apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts:221]: why don't we change the flow a bit more? 

```
compute()
if(result <= max) return
check constraints return
check feasibility return
```

I guess my concern is the case where `compute` is more expensive than the feasibility check.

Comment (eeeqeee) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts:221]: I tested both Hu Tao `8 N2C` and Cyno `4TF Quickbloom` and they were both faster in this PR. On which multitarget do you think `compute` is more expensive?

Comment (frzyc) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts:221]: Can you try the same with moving the constraint check after the max check? wonder if that is faster as well

**Review by frzyc — APPROVED** [review]

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts (hunk 1 above)
Comment (frzyc) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts]: To answer your previous question, I think we probably shouldn't cater to crazy mtargets, this solution is good.

**Pull request merged**